net/nginx-util/Makefile

```diff
@@ -0,0 +1,70 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=nginx-util
+PKG_VERSION:=1.0
+PKG_RELEASE:=1
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/nginx-util
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=Web Servers/Proxies
+  TITLE:=Builder of LAN listen directives for Nginx
+  DEPENDS:=+libstdcpp +libubus +libubox +libpthread
+  PROVIDES:=nginx-util
+endef
+
+define Package/nginx-ssl-util/default
+  $(Package/nginx-util)
+  TITLE+= and manager of its SSL certificates
+  DEPENDS+=+libopenssl
```

Comment:
Does this build without deprecated APIs enabled in openssl?

Reply:
Answered my own question. This needs headers:

```
0%] Building CXX object CMakeFiles/px5g.dir/px5g.cpp.o
In file included from /home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g.cpp:5:
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:22:65: error: 'RSA_F4' was not declared in this scope
   22 | auto gen_rsakey(const int keysize, const unsigned long exponent=RSA_F4);
      |                                                                 ^~~~~~
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp: In function 'auto gen_rsakey(int, long unsigned int)':
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:127:32: error: 'OPENSSL_RSA_MAX_MODULUS_BITS' was not declared in this scope; did you mean 'OPENSSL_ECC_MAX_FIELD_BITS'?
  127 |     if (keysize<512 || keysize>OPENSSL_RSA_MAX_MODULUS_BITS) {
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                OPENSSL_ECC_MAX_FIELD_BITS
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:133:19: error: 'BN_new' was not declared in this scope; did you mean 'BIO_new'?
  133 |     auto bignum = BN_new();
      |                   ^~~~~~
      |                   BIO_new
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:141:16: error: 'RSA_new' was not declared in this scope
  141 |     auto rsa = RSA_new();
      |                ^~~~~~~
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:144:14: error: 'BN_set_word' was not declared in this scope
  144 |         if (!BN_set_word(bignum, exponent) ||
      |              ^~~~~~~~~~~
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:145:14: error: 'RSA_generate_key_ex' was not declared in this scope
  145 |             !RSA_generate_key_ex(rsa, keysize, bignum, NULL))
      |              ^~~~~~~~~~~~~~~~~~~
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:147:13: error: 'RSA_free' was not declared in this scope
  147 |             RSA_free(rsa);
      |             ^~~~~~~~
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:152:5: error: 'BN_free' was not declared in this scope; did you mean 'BIO_free'?
  152 |     BN_free(bignum);
      |     ^~~~~~~
      |     BIO_free
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:165:9: error: 'RSA_free' was not declared in this scope
  165 |         RSA_free(rsa);
      |         ^~~~~~~~
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp: In function 'void selfsigned(const EVP_PKEY_ptr&, const char*, long unsigned int, const char*, bool)':
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:350:19: error: 'BN_new' was not declared in this scope; did you mean 'BIO_new'?
  350 |     auto bignum = BN_new();
      |                   ^~~~~~
      |                   BIO_new
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:354:31: error: 'BN_RAND_TOP_ANY' was not declared in this scope
  354 |     if (!BN_rand(bignum, 159, BN_RAND_TOP_ANY, BN_RAND_BOTTOM_ANY)) {
      |                               ^~~~~~~~~~~~~~~
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:354:48: error: 'BN_RAND_BOTTOM_ANY' was not declared in this scope
  354 |     if (!BN_rand(bignum, 159, BN_RAND_TOP_ANY, BN_RAND_BOTTOM_ANY)) {
      |                                                ^~~~~~~~~~~~~~~~~~
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:354:10: error: 'BN_rand' was not declared in this scope; did you mean 'srand'?
  354 |     if (!BN_rand(bignum, 159, BN_RAND_TOP_ANY, BN_RAND_BOTTOM_ANY)) {
      |          ^~~~~~~
      |          srand
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:355:9: error: 'BN_free' was not declared in this scope; did you mean 'BIO_free'?
  355 |         BN_free(bignum);
      |         ^~~~~~~
      |         BIO_free
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:360:9: error: 'BN_free' was not declared in this scope; did you mean 'BIO_free'?
  360 |         BN_free(bignum);
      |         ^~~~~~~
      |         BIO_free
/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/px5g-openssl.hpp:364:5: error: 'BN_free' was not declared in this scope; did you mean 'BIO_free'?
  364 |     BN_free(bignum);
      |     ^~~~~~~
      |     BIO_free
make[5]: *** [CMakeFiles/px5g.dir/build.make:63: CMakeFiles/px5g.dir/px5g.cpp.o] Error 1
make[5]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0'
make[4]: *** [CMakeFiles/Makefile2:111: CMakeFiles/px5g.dir/all] Error 2
make[4]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0'
make[3]: *** [Makefile:130: all] Error 2
make[3]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0'
make[2]: *** [Makefile:72: /home/mangix/devstuff/openwrt/build_dir/target-mips_24kc_musl/nginx-util-1.0/.built] Error 2
make[2]: Leaving directory '/home/mangix/devstuff/openwrt/feeds/packages/net/nginx-util'
time: package/feeds/packages/nginx-util/compile#0.53#0.08#0.61
make[1]: *** [package/Makefile:113: package/feeds/packages/nginx-util/compile] Error 2
make[1]: Leaving directory '/home/mangix/devstuff/openwrt'
make: *** [/home/mangix/devstuff/openwrt/include/toplevel.mk:227: package/nginx-util/compile] Error 2

```

Reply:
Ah, I included the rsa.h and bn.h, thank you :-)